analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs

```diff
@@ -67,10 +82,41 @@ private static void SerializeObjectToFile(string filePath, object objectToSerial
             using (var writer = XmlWriter.Create(stream, settings))
             {
                 var serializer = new XmlSerializer(objectToSerialize.GetType());
-                serializer.Serialize(writer, objectToSerialize, new XmlSerializerNamespaces(new[] {XmlQualifiedName.Empty}));
+                serializer.Serialize(writer, objectToSerialize, new XmlSerializerNamespaces(new[] { XmlQualifiedName.Empty }));
                 var ruleXml = Encoding.UTF8.GetString(stream.ToArray());
                 File.WriteAllText(filePath, ruleXml);
             }
         }
+
+#if NETCOREAPP
+
+        private static Type[] LoadAnalyzerTypes(string path) =>
+            Assembly.LoadFrom(path)  // We don't need 'Load', we have full control over the environment. It would make local build too complicated.
+                .ExportedTypes
+                .Where(x => !x.IsAbstract && typeof(DiagnosticAnalyzer).IsAssignableFrom(x))
+                .ToArray();
+
+        private static Rule[] LoadRules(Type[] analyzers) =>
```

Comment:
This whole logic is a reformulation of 
https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Utilities/RuleDetailBuilder.cs